optuna/cli.py

```diff
@@ -126,6 +126,16 @@ def get_parser(self, prog_name):
         parser.add_argument('--out', '-o',
                             help='Output HTML file path. If it is not given, a HTTP server starts '
                                  'and the dashboard is served.')
+        parser.add_argument('--allow-websocket-origin',
```

Comment:
@toshihikoyanase 
I have a question.
Is this test have to check can be access from remote host?

Reply:
I don't think so. Please check if dashboard command with --allow-websocket-origin  can be executed without any exceptions.
I think checking dashboard from remote hosts makes the test too complicated.

Reply:
I think too!
I will create test case thank you.

Reply:
@toshihikoyanase 

Sorry this comment is long.

I implemented two test patterns, which do you think is better?

## case1
```py3
@pytest.mark.parametrize('options', [['storage'], ['config'], ['storage', 'config'],
                                     ['storage', 'allow-websocket-origin1'],
                                     ['storage', 'allow-websocket-origin1', 'allow-websocket-origin2']])
def test_dashboard_command(options):
    # type: (List[str]) -> None

    with \
            StorageConfigSupplier(TEST_CONFIG_TEMPLATE) as (storage_url, config_path), \
            tempfile.NamedTemporaryFile('r') as tf_report:

        storage = RDBStorage(storage_url)
        study_name = storage.get_study_name_from_id(storage.create_new_study_id())

        command = ['optuna', 'dashboard', '--study', study_name, '--out', tf_report.name]
        command = _add_option(command, '--storage', storage_url, 'storage' in options)
        command = _add_option(command, '--config', config_path, 'config' in options)
        command = _add_option(command, '--allow-websocket-origin',
                              '192.168.0.1:5006', 'allow-websocket-origin1' in options)
        command = _add_option(command, '--allow-websocket-origin',
                              '192.168.0.2:5006', 'allow-websocket-origin2' in options)
        subprocess.check_call(command)

        html = tf_report.read()
        assert '<body>' in html
        assert 'bokeh' in html
```

## case2
```py3
@pytest.mark.parametrize('options', [['storage'], ['config'], ['storage', 'config']])
@pytest.mark.parametrize('num_allow_websocket_origin', [0, 1, 2])
def test_dashboard_command(options, num_allow_websocket_origin):
    # type: (List[str]) -> None

    with \
            StorageConfigSupplier(TEST_CONFIG_TEMPLATE) as (storage_url, config_path), \
            tempfile.NamedTemporaryFile('r') as tf_report:

        storage = RDBStorage(storage_url)
        study_name = storage.get_study_name_from_id(storage.create_new_study_id())

        command = ['optuna', 'dashboard', '--study', study_name, '--out', tf_report.name]
        command = _add_option(command, '--storage', storage_url, 'storage' in options)
        command = _add_option(command, '--config', config_path, 'config' in options)
        for i in range(num_allow_websocket_origin):
            command.extend(['--allow-websocket-origin', '192.168.0.{}:5006'.format(i+1)])
        subprocess.check_call(command)

        html = tf_report.read()
        assert '<body>' in html
        assert 'bokeh' in html
```

I think 'case1' is better because more simple.

Reply:
Thank you for your suggestion.

When I read the test cases in `test_cli.py`, I found test cases for optional arguments tend to be separated from those of mandatory arguments (e.g., [`test_create_study_command_with_study_name`](https://github.com/pfnet/optuna/blob/master/tests/test_cli.py#L88) and [`test_create_study_command_with_direction`](https://github.com/pfnet/optuna/blob/master/tests/test_cli.py#L119)).

How about leaving `test_dashboard_command` as it is and creating a separated test case for `--allow-websocket-origin` like `test_dashboard_command_with_allow_websocket_origin`?